en/micro-integrator/docs/setup/security/encrypting_plain_text.md

```diff
@@ -1,118 +1,151 @@
+    You will be prompted to provide details of the keystore that will be used for encryption.
 
-1. Create a new file in the MI_HOME directory. The file should be named according to your OS as explained below.
+    - Keystore location
+    - keystore type
+    - keystore alias
+    - Keystore password
 
-    * For Linux: The file name should be `password-tmp`.
-    * For Windows: The file name should be `password-tmp.txt`.
+4.  Execute one of the following commands to generate the secret:
 
-    !!! Note
-        When you start the server, the keystore password will be picked from this new file. Note that this file is automatically deleted from the file system after the server starts. Therefore, the admin has to create a new text file every time the server starts.
+    ```bash
+    # To encrypt secret and get output to console
+    mi secret create
 
-        Alternatively, if you want to retain the password file after the server starts, the file should be named as follows:
+    # To encrypt secret and get output to file
+    mi secret create file
 
-        * For Linux: The file name should be `password-persist`
-        * For Windows: The file name should be `password-persist.txt`
+    # To encrypt secret and get output as a .yaml file
+    mi secret create k8
 
-2. Add the primary keystore password (which is "wso2carbon" by default) to the new file and save. By default, the password provider assumes that both private key and keystore passwords are the same. If not, the private key password must be entered in the second line of the file.
+    # To bulk encrypt secrets defined in a properties file
+    mi secret create -f=</file_path>
+    ```
+
+### Using encrypted (dynamic) secrets
+
+To dynamically load secrets to server configurations, update the parameters in the `deployment.toml` file as an environment variable or system property:
+
+```toml tab='Environment Variable'
+[keystore.primary]
+password = "$env{ENV_VAR}"
+alias = "$env{ENV_VAR}"
+key_password = "$$env{ENV_VAR}"  
+
+[truststore]                  
+password = "$env{ENV_VAR}"
+```
 
-3. Now, start the server as a background process by running the following command.
-   ```bash
-   ./micro-integrator.sh start
-   ```
+```toml tab='System Property'
+[keystore.primary]
+password = "$sys{system.property}"
+alias = "$sys{system.property}"
+key_password = "$sys{system.property}"  
+
+[truststore]                  
+password = "$sys{system.property}"
+```
+
+Read more about [using dynamic server configurations](../../../setup/dynamic_server_configurations).
+
+To dynamically load secrets to synapse configurations, define the synapse secret as an environment variable.
+For example, consider the JMS user name and password in a proxy service as secrets:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<proxy name="JmsListner" pinnedServers="localhost" startOnLoad="true" transports="http https jms" xmlns="http://ws.apache.org/ns/synapse">
```

Comment:
This part is wrong. in synapse configs no secrets are always accessed with {wso2:vault-lookup('Alias')}
Example: https://docs.wso2.com/display/ESB480/Working+with+Passwords